services/user/Webmail/ui/src/main.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+import { createHashRouter, RouterProvider } from "react-router-dom";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { MilkdownProvider } from "@milkdown/react";
+import { ProsemirrorAdapterProvider } from "@prosemirror-adapter/react";
+
+import { Drafts, Editor, Home, Sent, Viewer } from "@routes";
+import { TooltipProvider } from "@shadcn/tooltip";
+
+import DefaultLayout from "./layouts/default";
+
+import "./styles/globals.css";
+
+const queryClient = new QueryClient();
+
+const router = createHashRouter([
```

Comment:
In the next version, we can use an SPA instead of a hash router, we just need to update the http server to return the same index for all requests, except for perhaps some whitelisted few API endpoints (like `messages` and the standard endpoints for `action_templates`, etc.)